### PR TITLE
Provides LookupValue to reference LookupTables from database entities.

### PR DESCRIPTION
**Pull request by andyHa**
LookupValue and LookupValues can be used to store string colums or
lists in database entities while verifying or normalizing them using
LookupTables.

### CODE REVIEW

File: src/main/java/sirius/biz/codelists/LookupValue.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+    private final String lookupTableName;
+    private LookupTable table;
+
+    private final Display display;
+    private final Export export;
+    private final CustomValues customValues;
+    private String value;
+
+    /**
```
Comment (idlira) [inline src/main/java/sirius/biz/codelists/LookupValue.java:29]: Just a design question: should those in-class enums be defined at the end of the class?

File: src/main/java/sirius/biz/codelists/README.md [hunk 2]
```diff
@@ -7,6 +7,9 @@ which provide access to more or less static data like "list of all known countri
 Such data can either be provided via [Jupiter](https://github.com/scireum/jupiter) or by an
 internal [CodeList](CodeLists.java).
 
+Not that [LookupValue](LookupValue.java) or [LookupValues](LookupValues.java) can be used to easily reference such
```
Comment (idlira) [inline src/main/java/sirius/biz/codelists/README.md:10]:
```suggestion
Note that [LookupValue](LookupValue.java) or [LookupValues](LookupValues.java) can be used to easily reference such
```

**Review by idlira — CHANGES_REQUESTED** [review]

File: src/main/java/sirius/biz/codelists/LookupValue.java (hunk 1 above)
Comment (andyHa) [reply src/main/java/sirius/biz/codelists/LookupValue.java:31]: we generally keep them at the top

File: src/main/java/sirius/biz/codelists/LookupValues.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        this.customValues = customValues;
+        this.display = display;
+        this.export = export;
+    }
+
+    /**
+     * Creates a new list using default settings.
+     * <p>
+     * Note that when using the list in database entities, the field has to be final, as the actual values
+     * is stored internally.
```
Comment (cschierle) [inline src/main/java/sirius/biz/codelists/LookupValues.java:55]:
```suggestion
     * are stored internally.
```

File: src/main/java/sirius/biz/util/Countries.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+     */
+    public static final String CODE_FIELD = "code";
+
+    /**
+     * Contains the two letter ISO country code.
+     */
+    public static final String FIELD_ISO2 = "mappings.isoAlpha2";
+
+    /**
+     * Contains th three letter ISO country code.
```
Comment (cschierle) [inline src/main/java/sirius/biz/util/Countries.java:50]:
```suggestion
     * Contains the three letter ISO country code.
```

**Review by idlira — APPROVED** [review]

File: src/main/java/sirius/biz/codelists/LookupValue.java (hunk 1 above)
Comment (sabieber) [reply src/main/java/sirius/biz/codelists/LookupValue.java:31]: Yes public ones at the top, private ones at the bottom in my opinion (as public ones are relevant for API and private ones are just implementation details like getters and setters are)

**Review by sabieber — APPROVED** [review]

File: src/main/java/sirius/biz/codelists/LookupValue.java [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+ * <p>
+ * Being a string value, {@link sirius.db.mixing.annotations.Length} and {@link sirius.db.mixing.annotations.NullAllowed}
+ * can be used as to further annotate a database field.
+ */
+public class LookupValue {
+
+    private final String lookupTableName;
+    private LookupTable table;
+
+    private final Display display;
```
Comment (jmuscireum) [inline src/main/java/sirius/biz/codelists/LookupValue.java:26]: display isn't used yet?

**Review by jmuscireum — APPROVED** [review]

**Pull request merged**